src/Uno.Toolkit.UI/Controls/LoadingView/CompositeLoadableSource.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Uno.Disposables;
+
+#if IS_WINUI
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Markup;
+#else
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Markup;
+#endif
+
+namespace Uno.Toolkit.UI
+{
+	/// <summary>
+	/// Represents an <see cref="ILoadable" /> aggregate that is <see cref="ILoadable.IsExecuting"/>
+	/// when any of its nested <see cref="Sources" /> is <see cref="ILoadable.IsExecuting"/>.
+	/// </summary>
+	[ContentProperty(Name = nameof(Sources))]
+	public partial class CompositeLoadableSource : FrameworkElement, ILoadable
+	{
+		public event EventHandler? IsExecutingChanged;
+
+		#region DependencyProperty: Sources
+
+		public static DependencyProperty SourcesProperty { get; } = DependencyProperty.Register(
+			nameof(Sources),
+			typeof(ObservableCollection<ILoadable>),
+			typeof(CompositeLoadableSource),
+			new PropertyMetadata(default(ObservableCollection<ILoadable>), (s, e) => ((CompositeLoadableSource)s).OnSourcesChanged(e)));
+
+		/// <summary>
+		/// Gets and sets the collection of nested <see cref="ILoadable" />.
+		/// </summary>
+		public ObservableCollection<ILoadable> Sources
+		{
+			get => (ObservableCollection<ILoadable>)GetValue(SourcesProperty);
+			set => SetValue(SourcesProperty, value);
+		}
+
+		#endregion
+		#region DependencyProperty: IsExecuting
+
+		public static DependencyProperty IsExecutingProperty { get; } = DependencyProperty.Register(
+			nameof(IsExecuting),
+			typeof(bool),
+			typeof(CompositeLoadableSource),
+			new PropertyMetadata(default(bool), (s, e) => ((CompositeLoadableSource)s).OnIsExecutingChanged(e)));
+
+		public bool IsExecuting
+		{
+			get => (bool)GetValue(IsExecutingProperty);
+			set => SetValue(IsExecutingProperty, value);
+		}
+
+		#endregion
+
+		private readonly ConcurrentDictionary<ILoadable, IDisposable> _sourceSubcriptions = new();
+		private readonly SerialDisposable _sourceCollectionSubscription = new();
+
+		public CompositeLoadableSource()
+		{
+			Sources = new ObservableCollection<ILoadable>();
+		}
+
+		private void OnSourcesChanged(DependencyPropertyChangedEventArgs e)
+		{
+			if (e.OldValue is ObservableCollection<ILoadable> oldCollection)
+			{
+				oldCollection.CollectionChanged -= OnSourcesCollectionChanged;
+				ClearSubscriptions();
+			}
+			if (e.NewValue is ObservableCollection<ILoadable> newCollection)
+			{
+				newCollection.CollectionChanged += OnSourcesCollectionChanged;
+
+				// there would be items already in the collection from xaml collection initializer (uno-specific)
+				AddSubscriptions(newCollection);
+			}
```

Comment:
This likely will create a leak. You'll need to make either sides weak, or handle loaded/unloaded